### PR TITLE
Add log uploader command line util

### PR DESCRIPTION
**Pull request by mjbvz**
Adds a new --upload-logs command line flag that allows users to upload log files from their current session to a secure endpoint.

These logs will only be uploaded after the user reviews them and confirms the upload. After confirmation, the uploaded logs will be zipped up and uploaded to a secure store that only vscode core team members have access to. When a user shares their log file id, we can then find their log files in order to investigate an issue

### CODE REVIEW

File: src/vs/code/electron-main/logUploader.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			host: endpoint.host,
+			path: endpoint.path,
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/zip',
+				'Content-Length': fs.statSync(logsPath).size
+			}
+		}, res => {
+			const chunks: (Buffer)[] = [];
+			res.on('data', (chunk: Buffer) => {
```
Comment (roblourens) [inline src/vs/code/electron-main/logUploader.ts]: I think requestService would do some of this for you

File: src/vs/code/electron-main/logUploader.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	if (await promptUserToConfirmLogUplod(logsPath)) {
+		const outZip = await zipLogs(logsPath);
+		const result = await postLogs(endpoint, logsPath, outZip);
+		console.log(localize('didUploadLogs', 'Uploaded logs ID: {0}', result.blob_id));
+	} else {
+		console.log(localize('userDeniedUpload', 'Canceled upload'));
+	}
+}
+
+async function promptUserToConfirmLogUplod(
```
Comment (roblourens) [inline src/vs/code/electron-main/logUploader.ts]: upload

File: src/vs/code/electron-main/logUploader.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+): TPromise<any> {
+	const endpoint = Endpoint.getFromProduct();
+	if (!endpoint) {
+		console.error(localize('invalidEndpoint', 'Invalid log uploader endpoint'));
+		return;
+	}
+
+	const logsPath = await channel.call('get-logs-path', null);
+
+	if (await promptUserToConfirmLogUplod(logsPath)) {
```
Comment (RMacfarlane) [inline src/vs/code/electron-main/logUploader.ts]: nit: typo

File: src/vs/code/electron-main/logUploader.ts (hunk 1 above)
Comment (mjbvz) [reply src/vs/code/electron-main/logUploader.ts]: Not sure we can pipe data into that though (`fs.createReadStream(outZip).pipe(req)`). Will take a look

**Pull request merged**